salt/utils/requisite.py

```diff
@@ -0,0 +1,470 @@
+"""
+The classes and functions in this module are related to requisite
+handling and ordering of chunks by the State compiler
+"""
+
+from __future__ import annotations
+
+import fnmatch
+import logging
+import sys
+from collections import defaultdict
+from collections.abc import Generator, Iterable, Sequence
+from enum import Enum, auto
+from typing import TYPE_CHECKING, Any
+
+import networkx as nx  # pylint: disable=3rd-party-module-not-gated
```

Comment:
nitpick: I don't think this is necessary anymore since `networkx` was added to Pylint's allowed third party modules.

```suggestion
import networkx as nx
```